pkg/collect/rook.go

```diff
@@ -0,0 +1,163 @@
+package collect
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type CmdResult struct {
+	Stdout []byte
+	Stderr []byte
+}
+
+type RookCmd struct {
+	Cmd      []string
+	Args     []string
+	Filename string // file in the support-bundle
+}
+
+var RookCmds = []RookCmd{
+	RookCmd{
+		Cmd:      []string{"ceph", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "status",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "fs", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "fs",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "fs", "ls"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "fs-ls",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "status"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-status",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "tree"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-tree",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "osd", "pool", "ls", "detail"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "osd-pool",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "health", "detail"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "health",
+	},
+	RookCmd{
+		Cmd:      []string{"ceph", "auth", "ls"},
+		Args:     []string{"-f", "json-pretty"},
+		Filename: "auth",
+	},
+}
+
+func Rook(ctx *Context) ([]byte, error) {
+	client, err := kubernetes.NewForConfig(ctx.ClientConfig)
+	if err != nil {
+		return nil, errors.New("Failed to create kubernetes clientset")
+	}
+
+	namespace := "rook-ceph"
+	selector := []string{"app=rook-ceph-operator"}
+	pods, podsErrors := listPodsInSelectors(client, namespace, selector)
+	if len(podsErrors) > 0 {
+		return nil, errors.New("Unable to find pod")
+	}
+
+	rookOutput := make(map[string][]byte)
+	for _, c := range RookCmds {
+		errCh := make(chan error, 1)
+		resultCh := make(chan CmdResult, 1)
+		go func() {
+			// TODO: find pod with 'Running' status, for now use first pod in the array
```

Comment:
Should we do this? When someone is troubleshooting a cluster, it feels like there's a chance that a pod is in a crashloopbackoff, and this could be relevant

Reply:
TODO means "I don't want to forget about this case and it will be eventually done (this commit or some future commit)".  

Now that I've finished a lot of the other testing, I came back around to this and handled this case.  See my latest commit.